Code/GraphMol/MolAlign/AlignMolecules.cpp

```diff
@@ -91,9 +94,28 @@ void getAllMatchesPrbRef(const ROMol &prbMol, const ROMol &refMol,
   if (matches.size() > 1e6) {
     std::string name;
     prbMol.getPropIfPresent(common_properties::_Name, name);
-    std::cerr << "Warning in " << __FUNCTION__ << ": " << matches.size()
-              << " matches detected for molecule " << name << ", this may "
-              << "lead to a performance slowdown.\n";
+    BOOST_LOG(rdWarningLog)
+        << "Warning in " << __FUNCTION__ << ": " << matches.size()
+        << " matches detected for molecule " << name << ", this may "
+        << "lead to a performance slowdown." << std::endl;
+  }
+  if (ignoreHs) {
+    // filter Hs out of the matches
+    for (auto &match : matches) {
+      auto tmatch = match;
+      match.clear();
+      std::copy_if(
+          tmatch.begin(), tmatch.end(), std::back_inserter(match),
+          [&prbMolForMatch](const std::pair<int, int> &mi) {
+            return prbMolForMatch.getAtomWithIdx(mi.first)->getAtomicNum() != 1;
+          });
+      // match.erase(std::remove_if(
+      //     match.begin(), match.end(),
+      //     [&prbMolForMatch](const std::pair<int, int> &mi) {
+      //       return prbMolForMatch.getAtomWithIdx(mi.first)->getAtomicNum() ==
+      //       1;
+      //     }));
```

Comment:
What's wrong with the erase? Also, why not C++20 `erase_if()`:
```suggestion
      match.erase_if(match, [&prbMolForMatch](const auto &mi) {
             return prbMolForMatch.getAtomWithIdx(mi.first)->getAtomicNum() ==1;
      }));
```

Reply:
> What's wrong with the erase? Also, why not C++20 `erase_if()`:

I had tried using `std::erase_if()` and had problems with it. Let's see if it works.